warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java

```diff
@@ -520,17 +533,79 @@ public void setupQuery(GenericQueryConfiguration genericConfig) throws Exception
             throw new QueryException("Did not receive a ShardQueryConfiguration instance!!");
         }
 
-        ShardQueryConfiguration config = (ShardQueryConfiguration) genericConfig;
+        if (this.federatedConfig != null) {
+            log.debug("Setting up federated query");
+            setupFederatedQuery();
+        } else {
+            log.debug("Setting up non-federated query");
+            ShardQueryConfiguration config = (ShardQueryConfiguration) genericConfig;
+            setupQuery(config);
+
+        }
+    }
+
+    private void setupFederatedQuery() {
+        List<ShardQueryConfiguration> configs = federatedConfig.getConfigs();
+        if (configs.size() == 1) {
+            setupQuery(configs.get(0));
+        } else {
+            ShardQueryConfiguration firstConfig = configs.get(0);
+            // If the first query could not be run, assume the remaining could not either. Return no results.
+            // TODO - Is this assumption correct? Should we check all sub-configs if any could be run?
+            if (!firstConfig.canRunQuery()) {
```

Comment:
For what this entails today, only calling 1 is probably ok.  However it might be more robust to call em all.  If one of them cannot run, then we can't run.